src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,70 @@
+package duke;
+
+/**
+ * Parser handles user input.
+ */
+public class Parser {
+    protected String input;
+    protected TaskList taskList;
+    protected Ui ui = new Ui();
+
+    /**
+     * Constructor that takes in user input and loaded task list.
+     *
+     * @param input
+     * @param taskList
+     */
+    public Parser(String input, TaskList taskList) {
+        this.input = input;
+        this.taskList = taskList;
+    }
+
+    /**
+     * Method to determine action based on user input.
+     */
+    public void readCommand() {
+        String arr[] = this.input.split(" ", 2);
+
+        if (input.equals("bye")) {
+            Storage storage = new Storage("../../../data/duke.txt");
```

Comment:
Same here. Instead of specifying it in multiple places, define it as a constant in a single class and use it